keras/optimizers.py

```diff
@@ -76,6 +76,10 @@ def get_updates(self, loss, params):
 
     def get_gradients(self, loss, params):
         grads = K.gradients(loss, params)
+        if None in grads:
+            raise ValueError('An operation has `None` for gradient :', params[grads.index(None)],
```

Comment:
`params[n]` probably doesn't add value here since there is not realistic way for users to be able to track what it stands for. Please remove.